src/js/services/highlightJSBadge.js

```diff
@@ -0,0 +1,71 @@
+window.onload = () => {
+  let codeSnippetButtons;
+
+  const options = {
+    // the selector for the badge template
+    templateSelector: '#CodeBadgeTemplate',
+
+    // base content CSS selector that is searched for snippets
+    contentSelector: 'body',
+
+    // Delay in ms used for `setTimeout` before badging is applied
+    // Use if you need to time highlighting and badge application
+    // since the badges need to be applied afterwards.
+    // 0 - direct execution (ie. you handle timing
+    loadDelay: 0,
+
+    // CSS class(es) used to render the copy icon.
+    copyIconClass: 'fa fa-copy',
+    // optional content for icons class (<i class="fa fa-copy"></i> or <i class="material-icons">file_copy</i>)
+    copyIconContent: '',
+
+    // CSS class(es) used to render the done icon.
+    checkIconClass: 'fa fa-check text-success',
+    checkIconContent: '',
+
+    // function called before code is placed on clipboard that allows you inspect and modify
+    // the text that goes onto the clipboard. Passes text and code root element (hljs).
+    // Example:  function(text, codeElement) { return text + " $$$"; }
+    onBeforeCodeCopied: function (text, codeElement) {
+      if (!codeSnippetButtons) {
+        codeSnippetButtons = document.getElementsByClassName('code-badge');
```

Comment:
no longer have to find this on every copy code click on a page. do it once per page and cache it.

Reply:
we use `code-badge` instead of `hljs` because it is useful to the `mousedown` darker background transitory state logic.